### PR TITLE
Docs/configuration: fix incorrect env-subst syntax

### PR DESCRIPTION
**Pull request by markspolakovs**
Signed-off-by: Marks Polakovs <marks.polakovs@couchbase.com>

**What this PR does / why we need it**:

The docs currently state that the syntax for default values if `config.expand-env=true` is `${VAR:default}`. However, Loki uses https://github.com/drone/envsubst, where the correct syntax is `${VAR:-n}` - `${VAR:foo}` is the syntax to "offset $VAR n characters from start"

**Checklist**
- [x] Documentation added
- [n/a] Tests updated
- [n/a] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Review by chaudum — APPROVED** [review]

**Review by KMiller-Grafana — APPROVED** [review]
Doc change LGTM.

**Review by kavirajk — APPROVED** [review]
thanks @markspolakovs

**Comment by kavirajk** [comment]
@markspolakovs Thanks for the PR. can you please sign our CLA? I can merge that once it's done.

**Comment by markspolakovs** [comment]
@kavirajk Thanks for the ping and sorry for the delay - I'm waiting for my employer's Legal to approve the CLA, once that's through I'll sign it right away!

**Comment by markspolakovs** [comment]
Converted to draft because I mis-clicked the button on the CLA screen to sign it before it had been formally approved - apologies for the confusion!

**Comment by markspolakovs** [comment]
@kavirajk should be good to merge this now, apologies for the delay!

**Comment by NickLarsenNZ** [comment]
This one got me, please merge it in :)

**Comment by KMiller-Grafana** [comment]
PR 6119 reimplemented this PR.  Closing in favor of that PR.

**Pull request closed without merging**
